### PR TITLE
Seeding refactor

### PR DESCRIPTION
**Pull request by bkwon94**
Changed file structure and refactored seeding and data gen process

### CODE REVIEW

File: db/generate.js [hunk 1]
```diff
@@ -10,8 +10,8 @@ const createFakeUser = () => ({
 const createFakeReview = () => ({
   user_id: faker.random.number({ 'min': 1, 'max': 10000000 }),
   book_id: faker.random.number({ 'min': 1, 'max': 10000000 }),
-  date: `${faker.date.month()} ${faker.random.number({ 'min': 1, 'max': 30 })}, ${(Math.floor(Math.random() * (2019 - 1996)) + 1996)}`,
-  review: faker.lorem.paragraphs(),
+  date: `${faker.date.month()} ${faker.random.number({ 'min': 1, 'max': 30 })}, ${faker.random.number({ 'min': 1996, 'max': 2019 })}`,
```
Comment (rikotacards) [inline db/generate.js:13]: Nice change using the random number generator from Faker.

File: db/postgreSQL/seeds/addReviews.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+exports.seed = async (knex) => {
+
+  console.log('Beginning review data seed...');
+  console.time('seed review time');
+  // query to be used to copy CSV file into table
+  const query = `COPY reviews(user_id, book_id, date, review, rating) FROM '/Users/briankwon/Desktop/hr-immersive/SDC/sdc-brian/reviews.csv' DELIMITER ',' CSV HEADER`;
+  let count = 0;
+  // function that deletes csv each iteration to avoid memory error
+  const deleteCSV = () => {
```
Comment (rikotacards) [inline db/postgreSQL/seeds/addReviews.js:13]: smart move here for conserving your computer's memory

File: db/postgreSQL/seeds/addReviews.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    fs.unlink('/Users/briankwon/Desktop/hr-immersive/SDC/sdc-brian/reviews.csv', err => {
+      if (err) {console.log(err)};
+    })
+  };
+  // count increment function
+  const inc = () => { count++; };
+  fs.createWriteStream('/Users/briankwon/Desktop/hr-immersive/SDC/sdc-brian/reviews.csv');
+  // truncate will empty the table if filled
+  knex.truncate('reviews');
```
Comment (rikotacards) [inline db/postgreSQL/seeds/addReviews.js:22]: heads up truncate or deleting rows from db won't reset index values in case you're using them as your product id.

**Review by rikotacards — APPROVED** [review]
Reviewed, left some comments, looks good to merge.

File: db/postgreSQL/seeds/addReviews.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+  const seedDB = async() => {
+    while (count < 50) {
+      let fakeReviews = [];
+      for (let i = 0; i < 1000000; i++) {
+        fakeReviews.push(generate.createFakeReview());
+      }
+      await new ObjectsToCsv(fakeReviews).toDisk('/Users/briankwon/Desktop/hr-immersive/SDC/sdc-brian/reviews.csv');
+      await knex.raw(query);
+      await deleteCSV();
+      await inc();
```
Comment (therobinkim) [inline db/postgreSQL/seeds/addReviews.js:33]: `inc()` seems like a fairly non-intuitive function name.

also, `inc` is synchronous, so I don't think we shouldn't be `await`ing it.

File: db/postgreSQL/seeds/addReviews.js [hunk 5]
```diff
@@ -0,0 +1,75 @@
+const generate = require('../../generate.js');
+const fs = require('fs');
+const ObjectsToCsv = require('objects-to-csv');
+
+exports.seed = async (knex) => {
+
+  console.log('Beginning review data seed...');
+  console.time('seed review time');
+  // query to be used to copy CSV file into table
+  const query = `COPY reviews(user_id, book_id, date, review, rating) FROM '/Users/briankwon/Desktop/hr-immersive/SDC/sdc-brian/reviews.csv' DELIMITER ',' CSV HEADER`;
```
Comment (therobinkim) [inline db/postgreSQL/seeds/addReviews.js:10]: I would recommend taking advantage of `__dirname` or the `path` module so you can remove these hardcoded paths that work for only your machine. (There are several other instances below as well.)

File: db/postgreSQL/seeds/addReviews.js [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+  // function that will seed db
+  const seedDB = async() => {
+    while (count < 50) {
+      let fakeReviews = [];
+      for (let i = 0; i < 1000000; i++) {
+        fakeReviews.push(generate.createFakeReview());
+      }
+      await new ObjectsToCsv(fakeReviews).toDisk('/Users/briankwon/Desktop/hr-immersive/SDC/sdc-brian/reviews.csv');
+      await knex.raw(query);
+      await deleteCSV();
```
Comment (therobinkim) [inline db/postgreSQL/seeds/addReviews.js:32]: I think we might be throwing `await` out in places where it doesn't do anything.. I would advise looking into how `async`/`await` work with ONLY promises and caution you against using them with callbacks (bc it probably isn't doing what you're hoping it does).

TLDR: `async`/`await` isn't magic and works in a very particular way. If you're not using them in a very particular way, you might be convincing yourself that you're doing something useful (when you're actually not).

File: .eslintrc.js [hunk 7]
```diff
@@ -1,7 +1,7 @@
...
-  }
-}
\ No newline at end of file
+// module.exports = {
+//   "extends": "airbnb",
+//   "env": {
+//     "jest": true,
+//     "browser": true
+//   }
+// }
```
Comment (therobinkim) [inline .eslintrc.js:7]: What happened to the eslint config? :(

File: db/postgreSQL/seeds/addReviews.js [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+  const query = `COPY reviews(user_id, book_id, date, review, rating) FROM '/Users/briankwon/Desktop/hr-immersive/SDC/sdc-brian/reviews.csv' DELIMITER ',' CSV HEADER`;
+  let count = 0;
+  // function that deletes csv each iteration to avoid memory error
+  const deleteCSV = () => {
+    fs.unlink('/Users/briankwon/Desktop/hr-immersive/SDC/sdc-brian/reviews.csv', err => {
+      if (err) {console.log(err)};
+    })
+  };
+  // count increment function
+  const inc = () => { count++; };
```
Comment (anacollado) [inline db/postgreSQL/seeds/addReviews.js:19]: It seems clearer to inline this incrementing instead of making it a function.
